modules/build/src/main/scala/scala/build/Project.scala

```diff
@@ -72,6 +72,21 @@ final case class Project(
   def bloopFile: BloopConfig.File =
     BloopConfig.File(BloopConfig.File.LatestVersion, bloopProject)
 
+  private def updateScalacOptions(scalacOptions: List[String]): List[String] = {
+    val keysWhichAcceptRelativePath = Seq(
+      "-coverage-out:"
+    )
+    def maybeConvertRelativePathToAbsolute(scalacOption: String): String =
+      keysWhichAcceptRelativePath.find(scalacOption.startsWith) match {
+        case Some(scalacKey) =>
+          val maybeRelativePath = scalacOption.stripPrefix(scalacKey)
+          val absolutePath      = os.Path(maybeRelativePath, Os.pwd)
+          s"$scalacKey$absolutePath"
+        case None => scalacOption
+      }
+    scalacOptions.map(maybeConvertRelativePathToAbsolute(_))
```

Comment:
I think we should keep grouping the options with `ShadowingSeq` / `ScalacOpt` just in case, so that the `-coverage-out:` is expected to be interpreted as a coverage option by scalac, not as a value of some other argument. Something like
```scala
ShadowingSeq.from(opts.map(ScalacOpt(_))).values.map { l =>
  // only look at the head, the tail is only values passed to it
  if (l.head.value.startsWith("-coverage-out:"))
    // actual -coverage-out: option
    ScalacOpt("-coverage-out:" + *absolute path*) +: l.tail
  else
    // not a -coverage-out: option
    l
}
```